packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -59,13 +66,58 @@ export async function tryHotswapDeployment(
 
 async function findAllHotswappableChanges(
   stackChanges: cfn_diff.TemplateDiff, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+  sdk: ISDK, rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedStackNames: {[key: string]: any}, hotswappableResources?: HotswapOperation[],
 ): Promise<HotswapOperation[] | undefined> {
   const resourceDifferences = getStackResourceDifferences(stackChanges);
 
   let foundNonHotswappableChange = false;
   const promises: Array<Array<Promise<ChangeHotswapResult>>> = [];
+
+  if (!hotswappableResources) {
+    hotswappableResources = new Array<HotswapOperation>();
+  }
+
+  // process any resources in nested stacks
+  // if any of them are not hotswappable, fail now
+  const resourceDifferenceEntries = Object.entries(resourceDifferences);
+  for (const [logicalId, change] of resourceDifferenceEntries.filter(
+    ([_, resourceDifference]) => (resourceDifference.newValue?.Type === 'AWS::CloudFormation::Stack' && resourceDifference.oldValue?.Type === 'AWS::CloudFormation::Stack'))
+  ) {
+    const parameters = change.newValue?.Properties?.Parameters;
+    if (parameters) {
+      for (const paramName in parameters) {
+        if (typeof parameters[paramName] !== 'string') {
+          parameters[paramName] = await evaluateCfnTemplate.evaluateCfnExpression(parameters[paramName]);
+
+        }
+      }
+    }
+
+    const nestedStackName = nestedStackNames[logicalId].stackName;
+    const evaluateNestedCfnTemplate = nestedStackName
+      ? evaluateCfnTemplate.createNestedEvaluateCloudFormationTemplate(nestedStackName,
+        sdk, rootStackArtifact, parameters)
+      : evaluateCfnTemplate;
```

Comment:
When can `nestedStackName` be `undefined`? And in that case, should we not just not attempt hotswapping at all? Why are we using the parent's `evaluateCfnTemplate` for the nested Stacks, even though we know it cannot possibly work?